src/Temporalio/Worker/ActivityWorker.cs

```diff
@@ -665,7 +673,7 @@ public override void Init(ActivityOutboundInterceptor outbound)
                 object? result;
                 try
                 {
-                    result = input.Delegate.DynamicInvoke(input.Parameters);
+                    result = input.Delegate.DynamicInvoke(input.Args);
                 }
                 catch (TargetInvocationException e)
                 {
```

Comment:
Not part of this PR, but is it correct to do `throw e.InnerException!` or is it better to do `ExceptionDispatchInfo.Capture(e.InnerException).Throw();`. More info: https://stackoverflow.com/a/17091351/2069

Reply:
Ah, missed this one (you'll see I use that `ExceptionDispatchInfo` approach in other places). Will fix.

Reply:
Fixed in next PR